jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5782,9 +5782,9 @@ public Set<AccessFlag> accessFlags() {
 	 */
 	public boolean isUnnamedClass() {
 		boolean rc = false;
-		if ((null == getEnclosingObjectClass())
+		if (isSynthetic()
+			&& (null == getEnclosingObjectClass())
 			&& (null == getDeclaringClass())
-			&& isSynthetic()
 			&& (Modifier.FINAL == (getModifiers() & Modifier.FINAL))
```

Comment:
Perhaps the `final` check should move up as well, merging with the modifiers access after the `!isArray()` test?
```java
		boolean rc = false;
		if (!isArray()) {
			int rawModifiers = getModifiersImpl();
			if ((Modifier.FINAL | SYNTHETIC) == (rawModifiers & (Modifier.FINAL | SYNTHETIC))) {
				if ((null == getEnclosingObjectClass()) && (null == getDeclaringClass())) {
					rc = true;
				}
			}
		}
		return rc;
```

Reply:
Yes, the final check can be moved up.

Reply:
Change is updated.